tests/test_adjust_fetch_data.c

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2018 Snowflake Computing, Inc. All rights reserved.
+ */
+#include <string.h>
+#include "utils/test_setup.h"
+
+/**
+ * Test SF_STMT_USER_REALLOC_FUNC, which reallocates a larger memory
+ * in case the given memory size is not large enough.
+ * @param unused
+ */
+void test_select_long_data_with_small_initial_buffer(void **unused) {
+    SF_CONNECT *sf = setup_snowflake_connection();
+    SF_STATUS status = snowflake_connect(sf);
+    if (status != SF_STATUS_SUCCESS) {
+        dump_error(&(sf->error));
+    }
+    assert_int_equal(status, SF_STATUS_SUCCESS);
+
+    /* query */
+    SF_STMT *sfstmt = snowflake_stmt(sf);
+
+    // realloc function is used to allocate a new memory when the given buffer
+    // size is not large enough to fit.
+    snowflake_stmt_set_attr(sfstmt, SF_STMT_USER_REALLOC_FUNC, realloc);
```

Comment:
This is the realloc for CAPI to call.

NOTE: we cannot assume CPI always use `realloc`, the standard C API function. The application may have its own memory functions, e.g., PHP ecalloc, erealloc, efree, so the function has to be specified by the application.